### PR TITLE
Add BitMEX.

### PR DESCRIPTION
**Pull request by STRML**
This implementation is not 100% ideal; BitMEX differs
from most markets in that we have a dynamic list of futures
expiries that are different from symbol to symbol.

In order to support this, the data persisted to Context
would need to include a parsed futures expiry, and we would
also need to support dynamic expiries per market. This is
[possible](https://github.com/BitMEX/BitcoinChecker/tree/bitmex-proper-futures), but may cause problems upstream.

In that interest, this commit presents the minimal changes
needed for usable BitMEX support.

This commit also orders markets alphabetically in development.

### CODE REVIEW

**Comment by MaciejCiemiega** [comment]
@STRML Please remove the sorting part. Sorting by name is done in the app part. This is just a data module providing a raw list of exchanges (in chronological order).

**Comment by STRML** [comment]
@MaciejCiemiega Happy to, but are you sure - I found it extremely difficult to test without the sorting, not knowing where in the list the new data module would show up. I find working with it this way much more intuitive.

**Comment by MaciejCiemiega** [comment]
@STRML The order is as in `MarketsConfig.java` file ("creation date" order). So if one write a new exchange it will always be at the end.
DataModuleTester is as simple as possible and there is no alphabetical sorting there - it is just raw testing tool. In real app there is a sorting by name. If you think it would make the DataModuleTester more convenient to use you can do a PR with that feature, but this is no a DataModule responsibility.
So in other words: in DataModuele order should be by "creation date" and it is an app job to sort it alphabetically (or present it differently to the user).

@STRML apart from that LGTM. I will merge it and will try to release it today:)

**Comment by STRML** [comment]
Sure, and that makes sense, it's just a pain to test while scrolling all the way down the list each time. If anything, we should just reverse it, newest module first, so it's always at the top.

**Comment by MaciejCiemiega** [comment]
@STRML That makes a lot of sense:) Done: 8a198fc3ce990802d67c39c3cc3af25a7e307905

**Comment by STRML** [comment]
:) And I was just about to push a PR for it. Thanks!
